### ISSUE TITLE
Slow keying of images with large metadata in Nuke 11.2

### ISSUE THREAD

**Issue by jonahfriedman** [Psyop/Cryptomatte#90]
For example, the bunny_CryptoObject.exr example image. This is reported to Foundry in ticket [41021](https://support.foundry.com/hc/en-us/requests/41021). 

```
Reproduction steps: 

1. Read in our example file for bunny_CryptoObject.exr. 
    It can be downloaded here: 
    https://github.com/Psyop/Cryptomatte/raw/master/sample_images/bunny_CryptoObject.exr 

2. Run the following Python in the Nuke script editor: 

import time

def my_test_knob_changed_event(node):
    start = time.time()
    md = node.metadata()
    print "read md in", time.time() - start, "seconds"

nuke.addKnobChanged(lambda: my_test_knob_changed_event(nuke.thisNode()), nodeClass='Grade')

3. Add a grade node to the read node, and move it around in the node graph. Moving the node triggers the knob changed event. This does not trigger the bug, but shows that metadata caching working. Something like this should be printed in the script editor, with very fast times: 
    read md in 0.00300002098083 seconds 
    read md in 0.00200009346008 seconds 

4. On the grade node, change a slider, such as the black point. It should lag very noticably and you should get this (very slow): 
    read md in 0.178999900818 seconds 
    read md in 0.171999931335 seconds

```

This bug is triggered just by looking at metadata in a knob changed event, so I'm unsure how to work around it, and I think the ball is in Foundry's court.

**Comment by jonahfriedman** [comment]
This now [TP 364183](https://support.foundry.com/hc/en-us/articles/360000848879-TP-364183-There-is-a-performance-loss-while-changing-values-of-a-node-when-multiple-node-metadata-actions-are-called-?flash_digest=c841a4f2d6db41726312b3729a4b3263d70827dc) in Foundry bug tracker. 

They've determined this has been an issue since 11.1v3, but was not in 11.1v2.

**Comment by jonahfriedman** [comment]
Since there was no movement on this bug, I'm working on a workaround. It's currently in the "metadatacache" branch, and it works by caching metadata (except for the manifest) on a hidden string knob. (Not yet released.)

**Comment by jonahfriedman** [comment]
The workaround for this is released in version 1.2.4. There is still no movement from Foundry on this one.

**Comment by jonahfriedman** [comment]
Closing this, as we've worked around this. Still no news about the underlying issue.
